src/main/java/com/snowflake/kafka/connector/internal/streaming/schemaevolution/iceberg/IcebergColumnTypeMapper.java

```diff
@@ -19,23 +18,25 @@
 import org.apache.kafka.connect.data.Time;
 import org.apache.kafka.connect.data.Timestamp;
 
-public class IcebergColumnTypeMapper extends ColumnTypeMapper {
+class IcebergColumnTypeMapper {
+
+  static IcebergColumnTypeMapper INSTANCE = new IcebergColumnTypeMapper();
```

Comment:
While I understand why it was introduced, I think we can do better. If the mapping is moved from the node class to some kind of factory service this singleton won't be needed.